CalibPPS/ESProducers/plugins/CTPPSInterpolatedOpticalFunctionsESSource.cc

```diff
@@ -61,26 +72,29 @@ std::shared_ptr<LHCInterpolatedOpticalFunctionsSetCollection> CTPPSInterpolatedO
   // get the input data
   LHCOpticalFunctionsSetCollection const &ofColl = iRecord.get(opticsToken_);
 
-  LHCInfo const &lhcInfo = iRecord.get(lhcInfoToken_);
+  auto lhcInfoCombined = LHCInfoCombined::createLHCInfoCombined<
+      CTPPSInterpolatedOpticsRcd,
+      edm::mpl::Vector<CTPPSOpticsRcd, LHCInfoRcd, LHCInfoPerFillRcd, LHCInfoPerLSRcd>>(
+      iRecord, lhcInfoPerLSToken_, lhcInfoPerFillToken_, lhcInfoToken_, useNewLHCInfo_);
 
   // is there anything to do?
-  if (currentDataValid_ && lhcInfo.crossingAngle() == currentCrossingAngle_)
+  if (currentDataValid_ && lhcInfoCombined.crossingAngle() == currentCrossingAngle_)
     return currentData_;
 
   // is crossing angle reasonable (LHCInfo is correctly filled in DB)?
-  if (lhcInfo.crossingAngle() == 0.) {
+  if (lhcInfoCombined.crossingAngle() == LHCInfoCombined::crossingAngleInvalid) {
```

Comment:
If I understand it correctly, the crossing angle cannot be negative.
If so, to avoid the risk of requesting an exact comparisons between two floats (which can differ each other only because of some numerical fluctuation) you could just check instead:
```suggestion
  if (lhcInfoCombined.crossingAngle() < 0.f) {
```
or even
```suggestion
  if (lhcInfoCombined.crossingAngle() < -0.5f) {
```

Reply:
@vavati says that `crossingAngle` was never negative but from next year on it will be possible, so checking if xangle < 0.f is not a good idea.
 -1 as `crossingAngleInvalid` isn't bad since if `crossingAngle` is negative and valid it won't greater than around -140

Reply:
I do agree that comparing floats might be problematic - maybe I should introduce a static bool method to LHCInfoCombined named isCrossingAngleValid? It would simply look like this:
```
bool isCrossingAngleValid (float crossingAngle){
  return !(crossingAngle >= crossingAngleInvalid-1e-6 && crossingAngle <= crossingAngleInvalid+1e-6)
}
```